### PR TITLE
SERVER-84402 Extend mongo-perf $in benchmarks for larger unindexed collection

### PR DESCRIPTION
**Pull request by henrinikku**
Thanks for submitting a PR to the Mongo-Perf repo. Please include the following fields (if relevant) prior to submitting your PR.

**Jira Ticket:** SERVER-84402

**Whats Changed:**
Extend the $in benchmarks that actually use Bonsai.

**Patch testing results:**
https://spruce.mongodb.com/version/65980e41306615f8ca59a3f9/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: testcases/simple_in_queries.js [hunk 1]
```diff
@@ -16,7 +16,9 @@ if (typeof(tests) !== "object") {
         if (collation) {
             collectionOptions.collation = collation;
         }
-        for (const [nameSuffix, size] of [["", 10], ["BigCollection", 10000]]) {
+
+        const collectionSizes = [["", 10], ["BigCollection", 10000], ["VeryBigCollection", 1e6]];
```
Comment (alyacb) [inline testcases/simple_in_queries.js:20]: Noting (as discussed offline) that this isn't applicable to Bonsai M2, but could be useful coverage in future.

**Review by alyacb — APPROVED** [review]

File: testcases/simple_in_queries.js [hunk 2]
```diff
@@ -151,6 +153,21 @@ if (typeof(tests) !== "object") {
...
+            nDocs: 1e6,
+            docs: function (i) {
+                return {x: 2 * Random.randInt(largeInArray.length * 10)};
+            },
+            op: {
+                op: "find",
+                query: {x: {$in: largeInArray}}
+            }
+        });
     };
```
Comment (alyacb) [inline testcases/simple_in_queries.js:171]: Actually- can we also extend UnindexedLargeInMatching and UnindexedLargeInUnsortedMatching as well with the Big/VeryBig collection sizes?

File: testcases/simple_in_queries.js [hunk 3]
```diff
@@ -188,41 +205,44 @@ if (typeof(tests) !== "object") {
...
+     * Adds three test cases for a $in query: One per a collection of 10, 10K, and 1M documents.
+     */
+    function addNonMatchingInTestCases({name, largeInArray}) {
+        for (const [nameSuffix, size] of [["", 10], ["BigCollection", 10000], ["VeryBigCollection", 1e6]]) {
+            addQueryTestCase({
+                name: name + nameSuffix,
+                tags: ["regression"],
+                nDocs: size,
+                docs: function (i) {
+                    return {x: 2 * Random.randInt(1000) + 1};
```
Comment (alyacb) [inline testcases/simple_in_queries.js:218]: We should use largeInArray.length here instead of hard-coding 1000

**Review by alyacb — CHANGES_REQUESTED** [review]

File: testcases/simple_in_queries.js (hunk 3 above)
Comment (henrinikku) [reply testcases/simple_in_queries.js]: Done

**Review by alyacb — APPROVED** [review]

**Pull request merged**